android/quest/src/test/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModelTest.kt

```diff
@@ -914,9 +917,52 @@ class QuestionnaireViewModelTest : RobolectricTest() {
       questionnaireViewModel.saveQuestionnaireResponse(questionnaire, questionnaireResponse)
     }
   }
+  @Test
+  fun testPerformExtractionOnSuccessShowsSuccessToast() {
+    val successMessage = context.getString(R.string.structure_success)
+    val context = mockk<Context>(relaxed = true)
+    questionnaireViewModel.performExtraction(
+      context,
+      questionnaire = Questionnaire(),
+      questionnaireResponse = QuestionnaireResponse()
+    )
+    coVerify { context.getString(R.string.structure_success) }
+    coVerify { context.showToast(successMessage) }
+  }
+
+  @Test
+  fun testPerformExtractionOnFailureShowsMissingStructureMapToast() {
+    val missingStructureMapExceptionMessage =
+      context.getString(R.string.structure_map_missing_message)
+    val context = mockk<Context>(relaxed = true)
+    val questionnaire = Questionnaire()
+    val questionnaireResponse = QuestionnaireResponse()
+
+    coEvery { questionnaireViewModel.retrieveStructureMapProvider() } throws
+      NullPointerException("StructureMap on Null Object")
+
+    questionnaireViewModel.performExtraction(context, questionnaire, questionnaireResponse)
+
```

Comment:
let's check that questionnaireResponse is properly filled when missing, that can be in a separate test